deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -56,18 +65,36 @@ def _build_model(self):
         # shape: (batch size, num_sentences, num_sentence_words, embedding size)
         sentences_embedding = self._embed_input(sentences_input)
 
-        # We encode the question embeddings with some encoder.
+        # We encode the question embedding with some more seq2seq layers
+        modeled_question = question_embedding
+        for i in range(self.num_seq2seq_layers):
+            hidden_layer = self._get_seq2seq_encoder(name="question_seq2seq_{}".format(i),
```

Comment:
Note that with these names, it is no longer possible to make this an actual siamese network - there is no way to make these the same encoder with what you've done here.  And I'm not sure how to make it so you can do that just with parameters in the json file...  Probably you'd have to just pass a flag to the model class that says whether to use common names for these hidden layers.